### PR TITLE
T1106899 - Toolbar - An item's content is not updated according to the React state if it is nested in a DataGrid component instance  in React 18

### PR DESCRIPTION
**Pull request by MikeVitik**
In new react 18 guards call early then switch is rendered. Move scedule guard timeout's after switch is rendered.

### CODE REVIEW

**Review by GoodDayForSurf — APPROVED** [review]

File: packages/devextreme-react/src/core/templates-renderer.tsx [hunk 1]
```diff
@@ -4,8 +4,12 @@ import * as React from 'react';
 
 import { TemplatesStore } from './templates-store';
 
-class TemplatesRenderer extends React.PureComponent<{ templatesStore: TemplatesStore }> {
+class TemplatesRenderer extends React.PureComponent<{
+  templatesStore: TemplatesStore;
+  children?: React.ReactNode;
```
Comment (IlyaKhD) [inline packages/devextreme-react/src/core/templates-renderer.tsx:9]: As far as I can see there are  no references to `children`. Possibly remove it?

File: packages/devextreme-react/src/core/component-base.ts [hunk 2]
```diff
@@ -94,12 +94,12 @@ abstract class ComponentBase<P extends IHtmlOptions> extends React.PureComponent
...
   public componentDidUpdate(prevProps: P): void {
     this._updateCssClasses(prevProps, this.props);
-
     const config = this._getConfig();
     this._optionsManager.update(config);
     if (this._templatesRendererRef) {
-      this._templatesRendererRef.scheduleUpdate(this.useDeferUpdateForTemplates);
+      this._templatesRendererRef.scheduleUpdate(this.useDeferUpdateForTemplates, scheduleGuards);
     }
+    unscheduleGuards();
```
Comment (IlyaKhD) [inline packages/devextreme-react/src/core/component-base.ts:102]: Does this triggers guards operation for all the components on a page? That would be not the most efficient thing.

File: packages/devextreme-react/src/core/options-manager.ts [hunk 3]
```diff
@@ -103,7 +114,8 @@ class OptionsManager {
 
     changedOptions.forEach(([name, value]) => {
       const currentPropValue = config.options[name];
-      if (config.options.hasOwnProperty(name) && currentPropValue !== value) {
+      if (Object.prototype.hasOwnProperty.call(config.options, name)
```
Comment (IlyaKhD) [inline packages/devextreme-react/src/core/options-manager.ts:117]: Could you explain why what this change made for?

Comment (MikeVitik) [reply packages/devextreme-react/src/core/options-manager.ts:117]: https://eslint.org/docs/latest/rules/no-prototype-builtins

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**